tests/test_utils/compat.py

```diff
@@ -0,0 +1,47 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+from importlib.metadata import version
+from typing import TYPE_CHECKING, Any, cast
+
+from packaging.version import Version
+
+from airflow.models import Operator
+
+try:
+    from airflow.models.errors import ParseImportError
+except ImportError:
+    from airflow.models.errors import ImportError as ParseImportError  # type: ignore[no-redef]
+
+from airflow import __version__ as airflow_version
```

Comment:
@jedcunningham -> as discussed, here I switched to `__version__` to check for compatibility.